src/domain/relayer/evm/validations.rs

```diff
@@ -0,0 +1,218 @@
+use thiserror::Error;
+
+use crate::{
+    models::{RelayerEvmPolicy, U256},
+    services::EvmProviderTrait,
+};
+
+#[derive(Debug, Error)]
+pub enum EvmTransactionValidationError {
+    #[error("Validation error: {0}")]
+    ValidationError(String),
+    #[error("Insufficient balance: {0}")]
+    InsufficientBalance(String),
+}
+
+pub struct EvmTransactionValidator {}
+
+impl EvmTransactionValidator {
+    pub async fn validate_sufficient_relayer_balance(
+        balance_to_use: U256,
+        relayer_address: &str,
+        policy: &RelayerEvmPolicy,
+        provider: &impl EvmProviderTrait,
+    ) -> Result<(), EvmTransactionValidationError> {
+        let balance = provider
+            .get_balance(relayer_address)
+            .await
+            .map_err(|e| EvmTransactionValidationError::ValidationError(e.to_string()))?;
+
+        let min_balance = U256::from(policy.min_balance);
+        let remaining_balance = balance.saturating_sub(balance_to_use);
+
+        if balance_to_use.is_zero() && balance < min_balance {
```

Comment:
What about inverting logic here to try to make it simpler

For example:
```

let required_balance = balance_to_use.saturating_add(min_balance);
    
    if balance < required_balance {
        return Err(EvmTransactionValidationError::InsufficientBalance(
         .....
        ));
    }
```

Reply:
I see. I believe that printing out all data in single sentence could be enough.

I see three conditions that are not very readable.

Fee free to double check can it be optimised to minimise number of if conditions.